great_expectations/data_context/data_context.py

```diff
@@ -3250,6 +3254,9 @@ def test_yaml_config(
 
         The returned object is determined by return_mode.
         """
+        if return_mode not in ["instantiated_class", "report_object"]:
+            raise ValueError(f"Unknown return_mode: {return_mode}.")
+
```

Comment:
Moved to the very top of the function to exit early and fast (no unnecessary work)